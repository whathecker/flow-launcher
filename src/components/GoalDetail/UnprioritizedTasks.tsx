import React, { useContext } from "react";
import { TasksContext } from "../../contexts/tasks";
import { StyleSheet } from "react-native";
import { View, Text } from "../Themed";
import { UnprioritizedTasksList, EmptyUnprioritizedTasks } from "./components";
import { Typography, Color } from "../../styles";

const UnprioritizedTasks: React.FC = () => {
  const { state } = useContext(TasksContext);

  return (
    <View style={styles.wrapper}>
      <View style={styles.headerAreaWrapper}>
        <Text
          lightColor={Color.light.labelOnBackgroundForRead}
          darkColor={Color.dark.labelOnBackgroundForRead}
          style={styles.headerText}
        >
          {"Recently Added Tasks"}
        </Text>
      </View>
      {state.unprioTasks.length > 0 ? (
        <UnprioritizedTasksList tasks={state.unprioTasks} />
      ) : (
        <EmptyUnprioritizedTasks />
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    width: "90%",
    borderWidth: 0.5,
    borderColor: Color.light.defaultBorder,
    borderRadius: 5,
    padding: 20,
  },
  headerAreaWrapper: {
    marginBottom: 10,
  },
  headerText: {
    ...Typography.h4,
    fontSize: 16,
  },
});

export default UnprioritizedTasks;
