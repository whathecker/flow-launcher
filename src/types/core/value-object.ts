export type PriorityTier = "n/a" | "highest" | "high" | "mid" | "low";

export type PriorityMeasure = "n/a" | "yes" | "no"; // TODO: consider to use MIXED type (boolean and null) instead?

export class Priority {
  constructor(
    public tier: PriorityTier,
    public importance: PriorityMeasure,
    public urgency: PriorityMeasure,
  ) {}
}
