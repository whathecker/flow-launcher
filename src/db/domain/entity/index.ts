import { Priority } from "../value-object";

export type ID = number | string;

export type Reminder = "daily" | "three_days" | "seven_days";

export class Goal {
  constructor(
    public id: ID,
    public motivation: string,
    public reminder: Reminder,
    public tasks: Task[],
  ) {}
}

export type TaskStatus = "open" | "finished";

export class Task {
  constructor(
    public title: string,
    public description: string,
    public status: TaskStatus,
    public priority?: Priority,
    public id?: ID,
  ) {}
}
