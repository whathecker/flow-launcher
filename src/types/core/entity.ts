import { Priority } from "./value-object";

export type ID = number | string;

export type GoalStatus = "open" | "finished";
export type Reminder = "daily" | "three_days" | "seven_days";

export class Goal {
  constructor(
    public status: GoalStatus,
    public title: string,
    public motivation: string,
    public reminder: Reminder,
    public tasks: Task[],
    public _id?: ID,
  ) {}
}

export interface IAddGoalInput {
  title: string;
  motivation: string;
  reminder: string;
}

export interface IAddTaskInput {
  title: string;
  description: string;
  goal_id: string;
}

export interface IFetchTasksInput {
  goal_id: string;
}

export type TaskStatus = "open" | "finished";

export class Task {
  constructor(
    public _id: ID,
    public goal_id: ID,
    public title: string,
    public status: TaskStatus,
    public description?: string,
    public priority?: Priority,
  ) {}
}
