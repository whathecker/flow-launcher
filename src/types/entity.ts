import { Priority } from "./value-object";

export type ID = number | string;

export type GoalStatus = "open" | "finished";
export type Reminder = "daily" | "three_days" | "seven_days";

export class Goal {
  constructor(
    public status: GoalStatus,
    public title: string,
    public motivation: string,
    public reminder: Reminder,
    public tasks: Task[] | [],
    public _id?: ID,
  ) {}
}

export interface IAddGoalInput {
  title: string;
  motivation: string;
  reminder: string;
}

export type TaskStatus = "open" | "finished";

export class Task {
  constructor(
    public _id: ID,
    public goal_id: ID,
    public title: string,
    public description: string,
    public status: TaskStatus,
    public priority?: Priority,
  ) {}
}
