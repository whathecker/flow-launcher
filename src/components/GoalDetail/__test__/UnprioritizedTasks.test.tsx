import * as React from "react";
import renderer from "react-test-renderer";
import UnprioritizedTasks from "../UnprioritizedTasks";
import { Task } from "../../../types/core/entity";

describe("Test UnprioritizedTasks component", () => {
  const dummayTasks: Task[] = [
    {
      _id: "sjdiodfi103jsldicsd",
      goal_id: "asjdklasf0qwdakddasd",
      title: "test tasks",
      status: "open",
    },
    {
      _id: "sjdiodfi103jsldicsdsd",
      goal_id: "asjdklasf0qwdakddasd",
      title: "test tasks -1",
      status: "open",
    },
    {
      _id: "sjdiodfi103jsldicsdasdadgd",
      goal_id: "asjdklasf0qwdakddasd",
      title: "test tasks -2",
      status: "open",
    },
    {
      _id: "sjdiodfi103jsldicsdadasdasfwefewvfwe",
      goal_id: "asjdklasf0qwdakddasd",
      title: "test tasks-3",
      status: "open",
    },
  ];

  it("Should renders UnprioritizedTasks correctly", () => {
    const tree = renderer
      .create(<UnprioritizedTasks tasks={dummayTasks} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
