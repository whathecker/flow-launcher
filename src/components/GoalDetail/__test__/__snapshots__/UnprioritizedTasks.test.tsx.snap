// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test UnprioritizedTasks component Should renders UnprioritizedTasks correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "borderColor": "#848484",
        "borderRadius": 5,
        "borderWidth": 0.5,
        "padding": 20,
        "width": "90%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "marginBottom": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#848484",
          },
          Object {
            "fontFamily": "Lexend-SemiBold",
            "fontSize": 16,
          },
        ]
      }
    >
      Recently Added Tasks
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 100,
          "justifyContent": "center",
          "padding": 20,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#554F4F",
          },
          Object {
            "color": "#848484",
            "fontFamily": "Lexend-Regular",
            "fontSize": 14,
          },
        ]
      }
    >
      Nothing there yet
    </Text>
  </View>
</View>
`;
